src/zope/component/tests/test__declaration.py

```diff
@@ -80,22 +80,11 @@ def _run_generated_code(self, code, globs, locs,
                             fails_under_py3k=True,
                            ):
         import warnings
-        #from zope.component._compat import PYTHON3
-        PYTHON3 = False
```

Comment:
For my own reference, because I'd never looked into this before: The `zope.interface` versions are broken because they want to modify the `__metaclass__` variable of the caller, which is what doesn't work on Python 3. They do this in order to arrange to be run *after* the class has been constructed and hence its bases determined, so that they can "flatten" all of the implements declarations from the parent classes into the child class, presumably for performance.

`z.c.adapts` doesn't have this issue, because it just simply assigns its argument to a value in the caller's locals. Parents aren't considered.